source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -198,6 +198,7 @@ FilterConfig::FilterConfig(
       stats_(FilterConfig::generateStats(stats_prefix, scope)),
       encoded_resource_query_params_(encodeResourceList(proto_config.resources())),
       forward_bearer_token_(proto_config.forward_bearer_token()),
+      preserve_authorization_header_(proto_config.preserve_authorization_header()),
```

Comment:
We should add a config validation for the case where both `forward_bearer_token` and `preserve_authorization_header` are set to true.

Reply:
Is that the correct place where you normally do config validations?

https://github.com/denniskniep/envoy/blob/5e015c531e6813a4379a8b8cbfb7dddbb4ae4ec8/source/extensions/filters/http/oauth2/config.cc#L67-L73